nats/stream_command.go

```diff
@@ -1042,6 +1046,41 @@ func (c *streamCmd) lsAction(_ *kingpin.ParseContext) (err error) {
 	return nil
 }
 
+func (c *streamCmd) lmsAction(_ *kingpin.ParseContext) (err error) {
+	_, err = prepareHelper("", natsOpts()...)
+	kingpin.FatalIfError(err, "setup failed")
+
+	c.stream, err = selectStream(c.stream, c.force)
+	kingpin.FatalIfError(err, "could not select Stream")
+
+	var (
+		stream *jsm.Stream
+		info *api.StreamInfo
+		msg *api.StoredMsg
+	)
+	stream, err = jsm.LoadStream(c.stream)
+	kingpin.FatalIfError(err, "could not request Stream info")
+
+	info, err = stream.LatestInformation()
+	if err != nil {
+		return err
+	}
+
+	for seq := int(info.State.FirstSeq); seq <= int(info.State.LastSeq); seq++ {
+		msg, err = stream.ReadMessage(seq)
```

Comment:
looks generally ok but I am quite reluctant to do this with this API, this will make potentially thousands of API requests, all producing audit logs and more.

I still think we need a API to read multiple messages at a time up to - eg. 250 a time - and then a tool like this can page through the results and specify things like starting sequence etc.

What do you think @derekcollison ?

Reply:
I think at minimum we split the PR into two. For the stream messages, I still think we can utilize an ephemeral shadow consumer to achieve what you want without additional APIs.

Reply:
I started with the consumer to start with and then moved to this approach. For the consumer, I was doing a newconsumer and then a delete, my concern and why I moved away was name conflict.  So when you say ephemeral, are you saying to create the consumer without a name and defer deletion?

Reply:
JetStream has a native concept of ephemral consumers where it will delete them for you after a while - dont give it a durable name.

With the consumer approach we can also have more flexibility around how to start - hour ago, specific mesage, only tail etc.

What would be your desired behaviour here - the current PR just tails forever and might just scroll scroll scroll, what about paging like unix `more` does - go down 50 messages a time? I am just curious what the actual use case is for the code as it stands

Reply:
We can't use ephemeral consumers with pull mode and push mode has no windowing, doing this with durables will be pretty sucky as it would be easy to leave random durables around as people interrupt the CLI unclean ways.

So I am leaning back to wanting a API to page a stream similar to `$JS.API.STREAM.MSG.GET` but that takes a count.

Reply:
I made a command `nats stream view` that looks a bit like this:

```
[rip@grit]% ./nats str view x --id 440 --user test --password test                                                                                                                                            <17:30:15
Subject: x Received: 2020-10-15 17:26:14.981081489 +0200 CEST

  Msg-Id: 21

msg 21

Subject: x Received: 2020-10-15 17:26:15.009517643 +0200 CEST

  Msg-Id: 22

msg 22

Subject: x Received: 2020-10-15 17:26:15.037295881 +0200 CEST

  Msg-Id: 23

msg 23

Subject: x Received: 2020-10-15 17:26:15.066872537 +0200 CEST

  Msg-Id: 24

msg 24

Subject: x Received: 2020-10-15 17:26:15.102089424 +0200 CEST

  Msg-Id: 25

msg 25

Subject: x Received: 2020-10-15 17:26:15.131733166 +0200 CEST

  Msg-Id: 26

msg 26

Subject: x Received: 2020-10-15 17:26:15.163476279 +0200 CEST

  Msg-Id: 27

msg 27

Subject: x Received: 2020-10-15 17:26:15.186756101 +0200 CEST

  Msg-Id: 28

msg 28

Subject: x Received: 2020-10-15 17:26:15.212508691 +0200 CEST

  Msg-Id: 29

msg 29

Subject: x Received: 2020-10-15 17:26:15.237989617 +0200 CEST

  Msg-Id: 30

msg 30

? Next Page? Yes
Subject: x Received: 2020-10-15 17:26:15.266916182 +0200 CEST

  Msg-Id: 31

msg 31

Subject: x Received: 2020-10-15 17:26:15.294871584 +0200 CEST

  Msg-Id: 32

msg 32

Subject: x Received: 2020-10-15 17:26:15.32623436 +0200 CEST

  Msg-Id: 33

msg 33

Subject: x Received: 2020-10-15 17:26:15.361865229 +0200 CEST

  Msg-Id: 34

msg 34

Subject: x Received: 2020-10-15 17:26:15.401547151 +0200 CEST

  Msg-Id: 35

msg 35

Subject: x Received: 2020-10-15 17:26:15.428199987 +0200 CEST

  Msg-Id: 36

msg 36

Subject: x Received: 2020-10-15 17:26:15.457020052 +0200 CEST

  Msg-Id: 37

msg 37

17:30:24 Reached apparent end of data
```

not amazing code wise, but I can merge this if you like.